### DISCUSSION TITLE
Textarea with dropdown (or multiline select)

### DISCUSSION THREAD

**Discussion by me21** [zauberzeug/nicegui#3359] [Q&A]
### Question

Hello,

I would like to display a multiline select control. By default its contents is single line. Basically I would like to replace the inner input control with a textarea. Is it doable? I found nothing in the NiceGUI and Quasar docs so far.

**Comment by python-and-novella** [comment]
Like this?
https://quasar.dev/vue-components/popup-edit#textarea-qeditor

**Comment by me21** [comment]
No, the behavior of Select component is almost fine. It displays a dropdown list, it has search as I type, the only thing that I need is line wrapping, so that if the selected option is really long, its text would wrap to the next line instead of scrolling out of view.

If I do

```python
myopts = {1: 'aaaaa', 2: 'aaaaaaaaaaaaaaaa', 3: 'aaaaaaa aaaaa aaaaaaaa aaaaaaa aaaaaaa aaaaaaaaaa'}
myselect = ui.select(myopts, value=1, new_value_mode='add-unique', on_change=lambda e: ui.notify(f'select value: {e.value}'))
with myselect.default_slot:
    ui.textarea('textarea').bind_value_from(myselect, backward=lambda v: myopts[v]).props('dense')
myselect2 = ui.select(myopts, value=1, new_value_mode='add-unique', on_change=lambda e: ui.notify(e.value)).bind_value(myselect)
```

the textarea is displayed next to the original input control. I would like to replace it instead.

**Reply by python-and-novella** [reply]
[falkoschindler](https://github.com/falkoschindler) answers it.
https://github.com/zauberzeug/nicegui/discussions/3342#discussioncomment-10022409

**Reply by me21** [reply]
Not exactly, he answers about wrapping the option text in the dropdown list. I need it to wrap in the select control when selected as well.

![изображение](https://github.com/user-attachments/assets/3b0ce165-ea3a-49c3-b4ae-5364ba9df268)

I would like the select control to display the selected option text like textarea does.

**Reply by python-and-novella** [reply]
![image](https://github.com/user-attachments/assets/1cf3028e-f99d-4d8b-a6dd-4d41546159c0)
```python3
ui.select([
    'Very long option that should wrap',
    'foo',
    'bar',
    'baz',
]).props('popup-content-class="max-w-[200px]"').classes('w-[200px]')
```

**Reply by me21** [reply]
Please try adding `new_value_mode='add-unique'` when constructing `select`. I think the wrapping won't work in that case.
